sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -603,6 +603,11 @@ functionArgs
 
 functionArg
     : expression
+    | nestedAllFields
```

Comment:
Why not just add support for the qualified name addition for `NestedAllFields` now? When we support a query like `SELECT message.* FROM BLAH;` Won't that require refactoring this change.